provider/kubernetes/kubernetes.go

```diff
@@ -308,6 +310,56 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 				}
 			}
 		}
+
+		for _, t := range i.Spec.TLS {
+			tlsSecret, exists, err := k8sClient.GetSecret(i.Namespace, t.SecretName)
+			if err != nil {
+				log.Errorf("Unable to retrieve secret %s/%s: %s", i.Namespace, t.SecretName, err)
+				continue
+			}
+			if !exists {
+				log.Warnf("Unable to find secret %s/%s", i.Namespace, t.SecretName)
+				continue
+			}
+
+			tlsCrtData, tlsCrtExists := tlsSecret.Data["tls.crt"]
+			tlsKeyData, tlsKeyExists := tlsSecret.Data["tls.key"]
+			if !tlsCrtExists || !tlsKeyExists {
+				if !tlsCrtExists {
+					log.Warnf("Secret %s/%s doesn't have an entry named 'tls.crt'", i.Namespace, t.SecretName)
+				}
+				if !tlsKeyExists {
+					log.Warnf("Secret %s/%s doesn't have an entry named 'tls.key'", i.Namespace, t.SecretName)
+				}
+				continue
+			}
+
+			var hosts []string
+			if len(t.Hosts) > 0 {
+				hosts = t.Hosts
+			} else {
+				// If no hosts are provided the hosts property should default ot the wildcard host. To ensure that the
+				// tls config only matches rules in this ingress, add all the used hosts to simulate a wildcard like
+				// behavior.
+				hosts = []string{}
+
+				for _, r := range i.Spec.Rules {
+					if r.Host != "" {
+						hosts = append(hosts, r.Host)
+					}
+				}
+			}
```

Comment:
My code coverage indicator tells me that the `else` branch does not have a corresponding test. We should change that.